klient/wait/conditions/conditions.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package conditions
+
+import (
+	"context"
+	batchv1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apimachinerywait "k8s.io/apimachinery/pkg/util/wait"
+	"log"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+)
+
+type Condition struct {
+	resources *resources.Resources
+}
+
+func New(resources *resources.Resources) *Condition {
+	return &Condition{resources: resources}
+}
+
+func (c *Condition) ResourceScaled(obj k8s.Object, scaleFetcher func(object k8s.Object) int32, replica int32) apimachinerywait.ConditionFunc {
+	return func() (done bool, err error) {
+		log.Printf("Checking if the resource %s/%s has been scaled to %d", obj.GetNamespace(), obj.GetName(), replica)
```

Comment:
These log statements will be a bit verbose when used.  I would consider removing them.

Reply:
Can I make these optional? By default it will be turned off but can be enabled by the users if required. Would that be ok ?

Reply:
I think outputting logs from the framework code could be an issue because, as a user, when I write tests, I don't expect extra output unless I use something like t.Log (etc).

As far as making it optional, here is what  we can do:
* Get this merged (without logs)
* Create a new PR to introduce more robust log capabilities via a level log framework 
* Then we can research for something that make sense for the project.
How does that sound ?